packages/server/graphql/types/JiraServerIntegration.ts

```diff
@@ -33,6 +35,21 @@ const JiraServerIntegration = new GraphQLObjectType<{teamId: string; userId: str
   name: 'JiraServerIntegration',
   description: 'Jira Server integration data for a given team member',
   fields: () => ({
+    id: {
+      type: new GraphQLNonNull(GraphQLID),
+      description: 'Composite key in jiraServer:providerId format',
+      resolve: async ({teamId, userId}: {teamId: string; userId: string}, _args, {dataLoader}) => {
+        const auth = await dataLoader
+          .get('teamMemberIntegrationAuths')
+          .load({service: 'jiraServer', teamId, userId})
+
+        if (!auth) {
+          return ''
+        }
+
+        return `jiraServer:${auth.providerId}`
```

Comment:
-2 The id is not unique, a user can have the same provider for multiple teams, each team has a different auth. Thus you need to use `jiraServer:${auth}.id` here.

The main issue here is that we're mixing the providers (which are per team and should be non-null) with the auth (which is per team member and should be nullable). So what we really should do is:
* make `JiraServerIntegration` nullable (also GitLab, AzureDevOps, MSTeams, ...)
* move `IntegrationProvider` fields to Team object.

No need to do it in this change, but we should address that.

Reply:
@Dschoordsch 
> a user can have the same provider for multiple teams
So in future provider will be added to the whole organisation?
Currently, I see that we have `teamId` column in the `IntegrationProvider` table.
Will we have `IntegrationProviderTeam` table with `providerId | teamId` columns?

Reply:
`auth` object does not have `id`. Made it like this: `return `jiraServer:${teamId}:${auth.providerId}``

Reply:
That's the current state, each "IntegrationProvider" has scope "org" or "team". Right now the organization is derived from the team, but we may store it separately #6014 at some point